components/arm/arm.go

```diff
@@ -355,6 +358,20 @@ func WrapWithReconfigurable(r interface{}, name resource.Name) (resource.Reconfi
 
 // Move is a helper function to abstract away movement for general arms.
 func Move(ctx context.Context, r robot.Robot, a Arm, dst spatialmath.Pose, worldState *referenceframe.WorldState) error {
+	joints, err := a.JointPositions(ctx, nil)
+	if err != nil {
+		return err
+	}
+	model := a.ModelFrame()
+	// check that joint positions are not out of bounds
+	_, err = motionplan.ComputePosition(model, joints)
+	if err != nil && strings.Contains(err.Error(), referenceframe.OOBErrString) {
+		err = errors.New("cartesian movements are not allowed when arm joints are out of bounds")
```

Comment:
Can we concatenate the returned `err` from above with this string?

Reply:
you got it. now the error statement is of form:
```
cartesian movements are not allowed when arm joints are out of bounds - joint A input out of bounds, input B needs to be within range [C D]
```